JavaScript/JavaScriptSDK/Context/Operation.ts

```diff
@@ -12,7 +12,10 @@ module Microsoft.ApplicationInsights.Context {
         public syntheticSource: string;
 
         constructor() {
-            this.id = Util.newGuid();
+            this.id = Util.newGuid();            
+            if (window && window.location && window.location.pathname) {
+                this.name = window.location.pathname;
```

Comment:
Good question. Currently in UI i see page URL and page name being treated separately - when you look at diagnostics search result for page view. Page name is usually a text description, more usage than diagnostics oriented. Where the url (and derived from it baseUrl) will arguably be more useful to classify the page view from dev perspective.
By default I think url will work better/will be more predictable than window.document.title used for default page name. This obviously can be overridden with a telemetry initializer.